src/clojurewerkz/elastisch/rest/document.clj

```diff
@@ -70,6 +70,19 @@
                                      index mapping-type id)
                {:body document :query-params (ar/->opts args)})))
 
+(defn upsert
+  "Updates an existing document in the search index with given partial document"
```

Comment:
This is copied from `update-with-partial-doc` without changes. Surely you want to mention the possibility of an insert?